src/targets/__tests__/maven.test.ts

```diff
@@ -338,3 +348,92 @@ describe('get gradle home directory', () => {
     expect(actual).toEqual(expected);
   });
 });
+
+describe('withGradleProps', () => {
+  beforeAll(() => {
+    setTargetSecretsInEnv();
+  });
+
+  afterAll(() => {
+    removeTargetSecretsFromEnv();
+  });
+
+  beforeEach(() => {
+    delete process.env.GRADLE_USER_HOME;
+  });
+
+  /**
+   * Checks whether the properties file is correct.
+   * @param path Path to the properties file.
+   */
+  async function testCorrectPropsFile(path: string): Promise<void> {
+    try {
+      const content = (await fsPromises.readFile(path)).toString();
+      expect(content).toMatch(`mavenCentralUsername=${DEFAULT_OPTION_VALUE}`);
+      expect(content).toMatch(`mavenCentralPassword=${DEFAULT_OPTION_VALUE}`);
+    } catch (error) {
+      throw new Error(`Cannot read the contents of the props file: ${error}`);
+    }
+  }
+
+  test('non-existent props file', async () => {
+    await withTempDir(async dir => {
+      process.env.GRADLE_USER_HOME = dir;
+      const expectedPropsPath = `${dir}/${GRADLE_PROPERTIES_FILENAME}`;
+
+      /**
+       * Expect 3 assertions:
+       *  2 testing whether the correct file in execution is created.
+       *  1 testing whether the user's file has been deleted.
+       */
+      expect.assertions(3);
+
+      const mvnTarget = createMavenTarget(getRequiredTargetConfig());
+      mvnTarget.upload = jest
+        .fn()
+        .mockImplementationOnce(
+          async () => await testCorrectPropsFile(expectedPropsPath)
+        );
+
+      await mvnTarget.publish('v3rs10n', 'r3v1s10n');
+      await expect(fsPromises.access(expectedPropsPath)).rejects.toThrowError(
+        /ENOENT: no such file/
+      );
+    });
+  });
+
+  test('existent props file', async () => {
```

Comment:
You are missing some test cases:
1. What if the callback function throws an error? You should test that we still restore the snapshot/delete our own props file
2. What if we simply cannot backup an existing snapshot file? Where's the test case for that?
3. What if we cannot restore a snapshot file for some reason?